// react-square-web-payments-sdk/dist/components/afterpay/afterpay.es.mjs
import * as e from "react";
import E, { useAfterpay as m } from "../../contexts/afterpay/afterpay.es.mjs";
import { AfterpayContext as x, default as z } from "../../contexts/afterpay/afterpay.es.mjs";
import { useEventListener as g } from "../../hooks/use-event-listener.es.mjs";
import { useForm as d } from "../../contexts/form/form.es.mjs";
import { ButtonLoader as k } from "./afterpay.styles.es.mjs";
function A({
  Button: t,
  buttonColor: n = "black",
  buttonType: o = "buy_now_with_afterpay",
  finalCtaButtonType: c = "buy_now",
  id: a = "rswps-afterpay-button",
  ...f
}) {
  const i = e.useRef(null), s = m(), { cardTokenizeResponseReceived: u } = d(), l = e.useMemo(
    () => ({
      buttonColor: n,
      buttonType: o,
      finalCtaButtonType: c,
      useCustomButton: Boolean(t)
    }),
    [t, n, o, c]
  ), w = async (p) => {
    if (p.stopPropagation(), !s) {
      console.warn("Afterpay/Clearpay button was clicked, but no Afterpay/Clearpay instance was found.");
      return;
    }
    try {
      const r = await s.tokenize();
      if (r.status === "OK")
        return u(r);
      let y = `Tokenization failed with status: ${r?.status}`;
      if (r && "errors" in r)
        throw y += ` and errors: ${JSON.stringify(r?.errors)}`, new Error(y);
      console.warn(y);
    } catch (r) {
      console.error(r);
    }
  };
  return e.useEffect(() => (s && s?.attach(`#${a}`, l), () => {
    s && s?.destroy();
  }), [s]), g({
    listener: w,
    type: "click",
    element: i,
    options: {
      passive: !0
    }
  }), t ? /* @__PURE__ */ e.createElement(t, { ...f, id: a, ref: i }) : /* @__PURE__ */ e.createElement("div", { ...f, id: a, ref: i }, s ? null : /* @__PURE__ */ e.createElement(k, null));
}
function C({
  badgeTheme: t = "black-on-mint",
  component: n,
  id: o = "rswps-afterpay-message",
  modalLinkStyle: c = "circled-info-icon",
  modalTheme: a = "mint",
  size: f = "md",
  ...i
}) {
  const s = e.useRef(null), u = m(), l = e.useMemo(
    () => ({
      badgeTheme: t,
      modalLinkStyle: c,
      modalTheme: a,
      size: f
    }),
    [t, c, a, f]
  );
  e.useEffect(() => {
    const r = async () => {
      await u?.attachMessaging(`#${o}`, l);
    };
    u && !n?.Message && r();
  }, [u, n?.Message, l]), g({
    listener: async (r) => {
      r.stopPropagation(), u?.displayInformationModal({ modalTheme: a });
    },
    type: "click",
    element: s,
    options: {
      passive: !0
    }
  });
  const p = n?.Message;
  return p ? /* @__PURE__ */ e.createElement(p, { ...i, id: o, ref: s }) : /* @__PURE__ */ e.createElement("div", { ...i, id: o });
}
function P({ includeBranding: t, id: n = "rswps-afterpay-widget", ...o }) {
  const c = e.useRef(null), a = m();
  return e.useEffect(() => {
    a && (async () => {
      await a?.attachCheckoutWidget(`#${n}`, {
        includeBranding: t
      });
    })();
  }, [a, t]), /* @__PURE__ */ e.createElement("div", { ...o, id: n, ref: c });
}
function $(t) {
  return /* @__PURE__ */ e.createElement(E, null, /* @__PURE__ */ e.createElement(A, { ...t }));
}
export {
  A as AfterpayButton,
  x as AfterpayContext,
  C as AfterpayMessage,
  z as AfterpayProvider,
  P as AfterpayWidget,
  $ as default
};
