// react-square-web-payments-sdk/dist/components/ach/ach.es.mjs
import * as e from "react";
import { useForm as x } from "../../contexts/form/form.es.mjs";
import { useEventListener as L } from "../../hooks/use-event-listener.es.mjs";
import { PayButton as y, SvgIcon as P } from "./ach.styles.es.mjs";
import { transformPlaidEventName as S } from "./ach.utils.es.mjs";
function T({
  accountHolderName: E,
  transactionId: b,
  callbacks: c,
  buttonProps: v,
  children: f,
  svgProps: R,
  redirectURI: _
}) {
  const [a, g] = e.useState(() => {
  }), [k, d] = e.useState(!1), { cardTokenizeResponseReceived: m, createPaymentRequest: s, payments: p } = x(), u = e.useRef(null), z = async (n) => {
    if (n.stopPropagation(), !a) {
      console.warn("ACH button was clicked, but no ACH instance was found.");
      return;
    }
    if (!s)
      throw new Error("`createPaymentRequest()` is required when using ACH payments");
    d(!0);
    try {
      const t = await a.tokenize({
        accountHolderName: E,
        intent: "CHARGE",
        amount: s.total.amount,
        currency: s.currencyCode
      });
      if (t?.status === "OK")
        return await m(t);
      let r = `Tokenization failed with status: ${t?.status ?? ""}`;
      if (t && "errors" in t)
        throw r += ` and errors: ${JSON.stringify(t?.errors)}`, new Error(r);
      console.warn(r);
    } catch (t) {
      console.error(t);
    } finally {
      d(!1);
    }
  };
  if (e.useEffect(() => {
    const n = new AbortController(), { signal: t } = n, r = async (l) => {
      const { tokenResult: o } = l.detail;
      o?.status == "OK" && await m(o);
    };
    return (async (l) => {
      const o = await p?.ach({
        transactionId: b
      }).then((w) => {
        if (!l?.aborted)
          return g(w), w;
      });
      l.aborted ? (o?.removeEventListener("ontokenization", r), await o?.destroy()) : o?.addEventListener("ontokenization", r);
    })(t), () => {
      n.abort();
    };
  }, [s, p]), c)
    for (const n of Object.keys(c))
      a?.addEventListener(
        S(n),
        c[n]
      );
  L({
    listener: z,
    type: "click",
    element: u,
    options: {
      passive: !0
    }
  });
  const { isLoading: C, ...h } = v ?? {}, i = C || !a || k;
  return f ? /* @__PURE__ */ e.createElement(y, { ...h, "aria-disabled": i, disabled: i, ref: u, type: "button" }, f) : /* @__PURE__ */ e.createElement(y, { ...h, "aria-disabled": i, disabled: i, ref: u, type: "button" }, /* @__PURE__ */ e.createElement(
    P,
    {
      fill: "none",
      height: "1em",
      viewBox: "0 0 36 24",
      width: "1em",
      xmlns: "http://www.w3.org/2000/svg",
      ...R
    },
    /* @__PURE__ */ e.createElement("rect", { fill: "url(#prefix__paint0_linear)", height: 24, rx: 4, width: 36 }),
    /* @__PURE__ */ e.createElement(
      "path",
      {
        clipRule: "evenodd",
        d: "M18.509 6.16a.89.89 0 00-1.018 0l-6.124 4.285a.848.848 0 00-.325.96.873.873 0 00.833.595h1.75v4.286h-1.75a.866.866 0 00-.875.857c0 .473.392.857.875.857h12.25a.866.866 0 00.875-.857.866.866 0 00-.875-.857h-1.75V12h1.75c.38 0 .717-.24.833-.596a.848.848 0 00-.324-.959L18.509 6.16zm2.116 10.126V12h-5.25v4.286h5.25zM18 7.91l3.395 2.376h-6.79L18 7.91z",
        fill: "#fff",
        fillRule: "evenodd"
      }
    ),
    /* @__PURE__ */ e.createElement("defs", null, /* @__PURE__ */ e.createElement("linearGradient", { gradientUnits: "userSpaceOnUse", id: "prefix__paint0_linear", x1: 36, x2: 0, y1: 12, y2: 12 }, /* @__PURE__ */ e.createElement("stop", { stopColor: "#01D09E" }), /* @__PURE__ */ e.createElement("stop", { offset: 1, stopColor: "#03E4AE" })))
  ), /* @__PURE__ */ e.createElement("span", null, "Pay with Direct debit (ACH)"));
}
export {
  T as Ach,
  T as default
};
